### ISSUE TITLE
No ability to lock record assignment 

### ISSUE THREAD

**Issue by greenlanegreb** [salesagility/SuiteCRM#7824]
I am raising this as high priority for the following reason: 

Within workflow, workflow rules will often trigger based on a change of assignment. If you are automating tasks, perhaps within a business, a user being able to change the allocation will either trigger inappropriate workflows or stop workflows from triggering entirely undermining the workflow functionality if there is no way to lock down the assignment and stop it from being edited. Turning off case edit functionality stops a major amount of things that should be able to be updated from being updated and is definitely not the answer.

**Comment by greenlanegreb** [comment]
I respectfully disagree that this has been marked down as a suggestion. I did emphasise above how this is something that can undermine client/customer management in terms of any escalations process where automated workflow can be interrupted by abuse of anybody being able to alter who the case was assigned to. 

In terms of there not being a history of who the case was assigned to, i'd imagine that the system taking this into account could actually be done through referencing the audit history and that this might be part of a fix in terms of addressing this issue. Many thanks.

**Comment by samus-aran** [comment]
The reasoning behind this been labelled as a suggestion is that it is a wider enhancement to the system. Simply put you are asking the ability to 'conditionally set fields to be read only' and that would be a larger task.

Yes we could ship a minimum solution where it would be only assigned user field but that would escalate quickly to be applied to all fields, combination or multiple fields etc.

Now regards to a user abusing the system due to changing the assigned user the same applies to 'any' field in the system and would still require the same wider solution which would be a suggestion rather than a bug.

Bugs are a break in existing functionality, suggestions are additional feature (regardless of how expected one may feel it is). Stating it as a suggestion does not de-value the importance just categorises it correctly.

**Comment by greenlanegreb** [comment]
Thanks for your message Ashley, I can see where you are coming from but I really do think assignment is actually different. The reason being is that assignment directly ties into automation and, incorrect changes to it can mess up the ability to trigger automated workflows. I'd also politely argue that assignment is actually really central to working with Suite CRM as different people have different roles and they are going to be crucial in terms of the way any business works. However, if somebody wanted to shirk work (in a business environment), they could simply change the assignment to stop any escalation or reporting stats from highlighting this. 

Also, with the issue being recorded as a suggestion, I would have thought there'd be no direct path for the functionality to be built in (from your stand point) or to be fixed (from my stand point). 

Thanks and Kind Regards.

**Comment by greenlanegreb** [comment]
Hi, are there any updates on this? Thanks ever so much.

**Comment by greenlanegreb** [comment]
Any updates on this? Thanks.
